### DISCUSSION TITLE
speed up stat calculation

### DISCUSSION THREAD

**Discussion by 733737** [OpenRCT2/OpenRCT2#16866] [Feature requests] [answered]
when you build a ride, it can take dozens of seconds to calculate stats. I do not see why the openrct2 developers would have a problem with making stat calculation instant, I think it should be doable for sure without impacting gameplay.

**Comment by duncanspumpkin** [comment]
Its a very expensive operation so that is why it is performed over a number of ticks.

**Comment by 733737** [comment]
is that so that on lower end computers the framerate doesn't drop?

**Reply by IntelOrca** [reply]
Yes, we haven't looked into increasing the number of rating cycles per tick. We might be able to, but it still needs to have a limit, so that the game remains deterministic and have balanced performance across all clients in multiplayer.

**Reply by 733737** [reply]
remaining deterministic wouldn't be an issue if the stat calculation is just done say 10x faster for everybody and not dependent on how fast each person's computer is

**Reply by IntelOrca** [reply]
I would need to profile rating calculation to see whether increasing the number of cycles performed by 10x is non-degrading.

**Reply by 733737** [reply]
could you at least double the stat calculation speed?

**Comment by 733737** [comment, marked as answer]
this has been improved by https://github.com/OpenRCT2/OpenRCT2/pull/19683
